### PR TITLE
Basicipcam

### LINKED ISSUE #33: IoT webcam honeypot
Depending on the specific webcam model chosen, any of these could be relevant:

CVE-2017-8224 - Backdoor account
CVE-2017-8222 - RSA key and certificates
CVE-2017-8225 - Pre-Auth Info Leak (credentials) within the custom http server
Authenticated RCE as root
Pre-Auth RCE as root
CVE-2017-8223 - Misc - Streaming without authentication
CVE-2017-8221 - Misc - "Cloud" (Aka Botnet)

You could choose any one model and see the relevant CVEs for it, let's say Provision PT-737.
Look at this exploit and its supported devices list to understand how widespread this is:
https://github.com/threat9/routersploit/blob/master/routersploit/modules/exploits/cameras/multi/P2P_wificam_rce.py

### PR DESCRIPTION
**Pull request by shay-ir**
closes #33

### CODE REVIEW

File: services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+            req_data = requests.get(self.image_src_url)
+            return req_data.content, req_data.headers["Content-Type"]
+        if self.image_src_path:
+            with open(self.image_src_path, "rb") as image_file_handle:
+                req_data = image_file_handle.read()
+                return req_data, DEFAULT_CONTENT_TYPE
+        return None
+
+    @staticmethod
+    def _get_post_redirect_target():
```
Comment (omercnet) [inline services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py]: looks more like a property than a method

File: services/ip_cam_trendnet_tv_ip100/www/Jview.html [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+       EmailImage.innerHTML = "[ON]"
+    }
+    else {
+       EmailImage.innerHTML = "[OFF]"
+    }
+}
+</SCRIPT>
+
+<body bgcolor=white onload="Init()">
+<form action="http://127.0.0.1/Jview.htm" method="POST">
```
Comment (omercnet) [inline services/ip_cam_trendnet_tv_ip100/www/Jview.html]: replace form action to `Jview.htm` to reduce fingerprintability

File: services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+ORIGINATING_IP_FIELD_NAME = "originating_ip"
+ORIGINATING_PORT_FIELD_NAME = "originating_port"
+REQUEST_FIELD_NAME = "request"
+DEFAULT_SERVER_VERSION = "Camera Web Server/1.0"
+CAMERA_IMAGE_PATH = "/image.jpg"
+
+DEFAULT_CONTENT_TYPE = "image/jpeg"
+
+
+class ThreadingHTTPServer(ThreadingMixIn, HTTPServer):
```
Comment (omercnet) [inline services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py]: please split the server and service logic to two separate files so easier to comprehend 
`ip_cam_service.py` and `ip_cam_server.py`

File: services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+from six.moves.urllib import parse as urlparse
+from six.moves.BaseHTTPServer import HTTPServer
+from six.moves.SimpleHTTPServer import SimpleHTTPRequestHandler
+from six.moves.socketserver import ThreadingMixIn
+
+import requests
+
+from honeycomb.servicemanager.base_service import ServerCustomService
+
+DEFAULT_PORT = 80
```
Comment (omercnet) [inline services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py]: if port parameter is required there's no need for DEFAULT_PORT

File: services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+
+    def test(self):
+        """Test service alerts and return a list of triggered event types."""
+        event_types = list()
+        self.logger.debug("executing service test")
+        # One alert for authorization attempt
+        requests.get("http://localhost:{}/content.html".format(self.service_args.get("port", DEFAULT_PORT)),
+                     headers={"Authorization": "username=\"test\""})
+        event_types.append(TRENDNET_ADMIN_ACCESS_EVENT)
+        # And one for POST
```
Comment (omercnet) [inline services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py]: you only need one test per alert_type

the way we test for alerts is by reading the logs top to bottom so we'll always hit the first alert and ignore the second one

File: services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+            self.wfile.write(image_data_content)
+        elif self.path.lower().startswith("/content.htm"):
+            authorization = self.headers.get("Authorization")
+            # Trying to connect to admin causes an alert
+            if authorization:
+                self.alert(self)
+            self.send_response(401)
+            self.send_header("WWW-Authenticate", "BASIC realm=\"Administrator\"")
+            self.end_headers()
+            self.wfile.write("Password Error. ")
```
Comment (omercnet) [inline services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py]: this is the error you normally get from this device on failed auth?

File: services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py (hunk 5 above)
Comment (frankiesimon) [reply services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py]: We will split up the behavior into two different events - stay tuned. :)

File: services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py (hunk 3 above)
Comment (shay-ir) [reply services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py]: That's not our code, thats how it is in the simple_http service

File: services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py (hunk 4 above)
Comment (shay-ir) [reply services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py]: It is needed for testing

File: services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py (hunk 3 above)
Comment (omercnet) [reply services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py]: yes, simple_http is a basic example, but look at more complex services (pretty much any other service actually) and you'll see a separation between server and service

it just makes life easier for others reading your code

File: services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py (hunk 4 above)
Comment (omercnet) [reply services/ip_cam_trendnet_tv_ip100/ip_cam_trendnet_tv_ip100_service.py]: the port in test args is hardcoded so the port in `test()` can also be hardcoded
or change the const name to TEST_PORT

File: services/ip_cam_trendnet_tv_ip100/test.args.json [hunk 7]
```diff
@@ -0,0 +1 @@
+{"port": 8080, "image_src_url": "http://farm4.static.flickr.com/3119/2692712013_d34b1de6f0_z.jpg"}
```
Comment (omercnet) [inline services/ip_cam_trendnet_tv_ip100/test.args.json]: test args port is 8080 but DEFAULT_PORT is 80
how is the test passing?? o_O

**Comment by omercnet** [comment]
see https://codecov.io/gh/Cymmetria/honeycomb_plugins/pull/66/diff
`send_head()` is never reached during test, is that expected?

**Pull request merged**